thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/auto/onboard/AutoOnboardPinotMetadataSource.java

```diff
@@ -154,7 +158,7 @@ private boolean shouldRemoveDataset(DatasetConfigDTO datasetConfigDTO, Set<Strin
       List<MetricConfigDTO> metrics = metricDAO.findByDataset(datasetConfigDTO.getDataset());
       int metricCount = metrics.size();
       for (MetricConfigDTO metric : metrics) {
-        if (!metric.isDerived()) {
+        if (!metric.isDerived() && !metric.getName().equals(ROW_COUNT)) {
```

Comment:
Are you planning to explicitly add a column called ROW_COUNT for all datasets?

Reply:
No, won't add a column. In ThirdEye, we will map "ROW_COUNT" to a count(*) metric. 
This line is to ensure that the metric won't be removed automatically.

Reply:
Thanks for the clarification.